src/node_modules/app/components/ScreenGauge.svelte

```diff
@@ -0,0 +1,136 @@
+<script context='module'>
+	import {writable} from 'svelte/store';
+
+	export const screenGauge = writable();
+</script>
+<script>
+	import {onMount, /* createEventDispatcher */} from 'svelte';
+	import * as _ from 'lamb';
+
+	// TODO Support per-instance stores AND/OR events too?
+	// Possibly useful for managing multiple windows.
+
+	export let fontSize;
+	export let bands = [40, 60, 80, 100];
+	export let devMode = false;
+
+	let mounted = false;
+	let sampleWidth, sampleHeight;
+
+	function computeDimensions (width, height) {
+		const display = {
+			width: window.innerWidth,
+			height: window.innerHeight,
+			orientation: window.screen.orientation,
+			pixelRatio: window.devicePixelRatio,
+			aspect: window.innerWidth / window.innerHeight,
+		};
+		const glyph = {
+			width,
+			height,
+		}
+		const text = {
+			width: display.width / glyph.width,
+			height: display.height / glyph.height,
+		}
+		return {
+			display,
+			glyph,
+			text,
+		};
+	}
+
+	function computeFlags (dimensions) {
+		return {
+			...dimensions,
+			size: {
+				xSmall: dimensions.text.width < bands[0],
+				small: true,
+				medium: dimensions.text.width >= bands[1],
+				large: dimensions.text.width >= bands[2],
+				xLarge: dimensions.text.width >= bands[3],
+			},
+			orientation: {
+				portrait: dimensions.display.aspect < 1,
+				landscape: dimensions.display.aspect >= 1,
+			},
+		}
+	}
+
+	const getClasses = _.pipe([
+		_.pairs,
+		_.filterWith(_.getAt(1)),
+		_.mapWith(_.getAt(0)),
+		_.joinWith(' ')
+	]);
```

Comment:
Here we could use [`getTruthyValuesKeys`](https://nestauk.github.io/svizzle/jsDocBuild/module-@svizzle_utils_object-array.html#.getTruthyValuesKeys):

```js
const getClasses = _.pipe([getTruthyValuesKeys, _.joinWith(' ')]);
```